src/backend/common/defines.hpp

```diff
@@ -49,6 +49,13 @@ typedef enum {
     AF_BATCH_DIFF,             /* signal and filter have different batch size */
 } AF_BATCH_KIND;
 
+enum class kJITHeuristics {
+    PASS               = 0, /* no eval necessary */
+    TREE_HEIGHT        = 1, /* eval due to jit tree height */
+    KERNEL_PARAM_SIZE  = 2, /* eval due to many kernel parameters */
+    MEM_PRESSURE       = 3  /* eval due to memory pressure */
+};
```

Comment:
I think enum class values should be `CamelCase` since they need `name::` resolution anyway while being used. It results in less characters and easy to read names like the following, hence better in my opinion.

```cpp
enum class kJITHeuristics : int {
    Pass = 0,
    TreeHeight,
    KernelParameterSize,
    MemoryPressure
};
```

Reply:
Done! I actually like the CamelCase for these, it does look better.